packages/rocketchat-lib/server/lib/interceptDirectReplyEmails.js

```diff
@@ -0,0 +1,124 @@
+const IMAP = require('imap');
+
+class IMAPIntercepter {
+	constructor() {
+		this.imap = new IMAP({
+			user: RocketChat.settings.get('Direct_Reply_Username'),
+			password: RocketChat.settings.get('Direct_Reply_Password'),
+			host: RocketChat.settings.get('Direct_Reply_Host'),
+			port: RocketChat.settings.get('Direct_Reply_Port'),
+			debug: RocketChat.settings.get('Direct_Reply_Debug') ? console.log : false,
+			tls: !RocketChat.settings.get('Direct_Reply_IgnoreTLS'),
+			connTimeout: 30000,
+			keepalive: true
+		});
+	}
+
+	openInbox(Imap, cb) {
+		Imap.openBox('INBOX', false, cb);
```

Comment:
Won't work directly. Would have to use like this https://github.com/RocketChat/Rocket.Chat/pull/7342/files#diff-2bf49456cc2b2e1c64643c6b70c38e7cR23 .So its better to pass, I think?